chore: Replace use of flakeheaven with Ruff for flake8-return

nkclemson opened the pull request:
Related:

- Closes #8457 
- https://github.com/meltano/meltano/issues/7383

```
ruff check --select=RET --add-noqa src tests
```

Question: Which RET error code should we consider, and which one should we put on the ignore list? For eg: RET505 to RET508 are the return type we consider more pythonic.

https://pypi.org/project/flake8-return/

<img width="788" alt="flake8_return" src="https://github.com/meltano/meltano/assets/83767876/31cfe30d-eafa-41b8-9f1d-0c5ef1256a30">

Commit:
ruff flake8-return integration #8457

nkclemson committed:
Merge branch 'meltano:main' into ruff_return_8457

edgarrmondragon commented on .pre-commit-config.yaml:
```diff
@@ -101,6 +101,7 @@ repos:
           - flake8-polyfill==1.0.2
           - flake8-print==5.0.0
           - flake8-quotes==3.3.1
+          - flake8_return==1.2.0
```
I don't think we want to add more flakeheaven plugins

```suggestion
```

edgarrmondragon commented on .pre-commit-config.yaml:
```diff
@@ -101,6 +101,7 @@ repos:
           - flake8-polyfill==1.0.2
           - flake8-print==5.0.0
```
We can also get rid of this

```suggestion
```

edgarrmondragon commented:
> Question: Which RET error code should we consider, and which one should we put on the ignore list? For eg: RET505 to RET508 are the return type we consider more pythonic.

Let me think about it. Regardless, we should probably look at the astral/Ruff docs instead.

nKandel committed:
removed flake8-return and flake8-print from flake8 plugin

Commit:
simplified code to fix WPS361 and RET503

Commit:
Merge branch 'ruff_return_8457' of github.com:nkclemson/meltano into ruff_return_8457

edgarrmondragon committed:
Fix some low-hanging fruit

* Ignored most RET501 and RET503 cause they might change the
  _semantics_ the developer originally was going for, even if addressing
  the violations is preferrable and functionally identical.
* Addressed most other easy fixes for RET504, RET505, RET506
* Fixed a few typos in those files

edgarrmondragon reviewed (approved):
Most RET50x were easily fixable, so I went ahead and pushed 82c44a7. Thanks @nkclemson!
